.github/workflows/ci-publish.yml

```diff
@@ -0,0 +1,127 @@
+# Copyright 2022 JanusGraph Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: CI Release artifacts publish
+
+on:
+  push:
+    tags:
+      - v*.*.*
+
+jobs:
+  build-all:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'zulu'
+          java-version: '8.0.312+7'
```

Comment:
It would be great, if you add default build-all to this script otherwise, you can break the cache. It wasn't easy to get the cache working correctly.

Reply:
I used steps from the RELEASE.md. 
We have 2 steps there: `install` and `deploy`. I will check tomorrow if we can apply `build-all`. I don't know what that means yet

Reply:
You just have to copy the setup from a other workflow and set as a dependency. https://github.com/JanusGraph/janusgraph/blob/master/.github/workflows/ci-release.yml#L44-L57 and https://github.com/JanusGraph/janusgraph/blob/master/.github/workflows/ci-release.yml#L61

Reply:
If speak about cache, i thought about github action cache.

Reply:
I see. I will need to double check if that's correct to do for a release and whats the difference if we don't include this option. At this moment I just followed the RELEASE.md steps. 
We don't have releases too often, so, I'm not sure if we need to worry about cache. Also, I don't fully understand the cache problem st this moment. Do we have cache misses and re-download something or is there some other problem with cache?

Reply:
Wo could remove cache fully for the release build in this case we get a cleanup each time. 

The problem is more about how the GitHub cache works and the space we have. It is pretty tight for our build to cache everything.

Reply:
`install` + `deploy` with cache takes about 5 minutes. Without cache takes about 10 minutes. I believe for release process it's totally fine to disable cache for this action. Thus, I'm disabling cache for this action.